src/applications/disability-benefits/686c-674/config/address-schema.js

```diff
@@ -0,0 +1,244 @@
+import { isChapterFieldRequired } from './helpers';
+
+const MILITARY_STATES = Object.entries(ADDRESS_DATA.states).reduce(
+  (militaryStates, [stateCode, stateName]) => {
+    if (ADDRESS_DATA.militaryStates.includes(stateCode)) {
+      return {
+        ...militaryStates,
+        [stateCode]: stateName,
+      };
+    }
+    return militaryStates;
+  },
+  {},
+);
+
+const USA = {
+  value: 'USA',
+  name: 'United States',
+};
+
+const militaryBaseInfo = () => (
+  <div className="vads-u-padding-x--2p5">
+    <AdditionalInfo
+      status="info"
+      triggerText="Learn more about military base addresses"
+    >
+      <span>
+        The United States is automatically chosen as your country if you live on
+        a military base outside of the country.
+      </span>
+    </AdditionalInfo>
+  </div>
+);
+
+export const addressSchema = {
+  type: 'object',
+  properties: {
+    'view:livesOnMilitaryBase': {
+      type: 'boolean',
+    },
+    'view:livesOnMilitaryBaseInfo': {
+      type: 'object',
+      properties: {},
+    },
+    countryName: {
+      type: 'string',
+      enum: countries.map(country => country.label),
+      default: USA.name,
+    },
+    addressLine1: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    addressLine2: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    addressLine3: {
+      type: 'string',
+      minLength: 1,
+      maxLength: 100,
+      pattern: '^.*\\S.*',
+    },
+    city: {
+      type: 'string',
+    },
+    stateCode: {
+      type: 'string',
+      enum: states50AndDC.map(state => state.value),
+      enumNames: states50AndDC.map(state => state.label),
+    },
+    province: {
+      type: 'string',
+    },
+    zipCode: {
+      type: 'string',
+      pattern: '^\\d{5}$',
+    },
+    internationalPostalCode: {
+      type: 'string',
+    },
+  },
+};
+
+/*
+This method is resusable with the 686c-674, but could probably be extended for resuse elsewhere. 
+chapterKey is used uniquely inside the 686c to check for workflow requirements on individual fields.
+schemaKey is the corresponding schema property name for address. 
+*/
+export const addressUISchema = (schemaKey, chapterKey) => ({
+  'view:livesOnMilitaryBase': {
+    'ui:title':
+      'I live on a United States military base outside of the United States',
+  },
+  'view:livesOnMilitaryBaseInfo': {
+    'ui:description': militaryBaseInfo,
+  },
+  countryName: {
+    'ui:required': formData => isChapterFieldRequired(formData, chapterKey),
+    'ui:title': 'Country',
+    'ui:options': {
+      updateSchema: (formData, schema, uiSchema) => {
+        const countryUI = uiSchema;
+        const countryFormData = formData[`${schemaKey}`];
+        if (formData[`${schemaKey}`]['view:livesOnMilitaryBase']) {
+          countryUI['ui:disabled'] = true;
+          countryFormData.countryName = USA.name;
```

Comment:
I know this probably isn't ideal to do...but for selecting the checkbox for living on a military base doesn't update the value for countryName, and setting a default as done right below doesn't either. 🤷‍♂

Reply:
I ran into this exact situation when I was implementing this in the 526 form. I also used this same strategy ;) Does the way you are doing this cause the State and Postal fields to appear? I'm having an issue with mine where using this approach doesn't trigger the updateSchema function of those fields, so they stay hidden from when the user had a non-USA country selected

Reply:
I added an extra check to state and postal fields that checks the value of the checkbox before checking the country field. It isn't that pretty, but it addresses the issue of going from non-USA country to USA military base.